docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -435,6 +431,32 @@ CommandResponseTopicPrefix = "edgex/command/response/"  # publishing outbound co
 !!! Note
         Most of the device service configuration is existing based on its need to already communicate with the message bus for publishing events/readings.  The last two lines are added to allow device services to subscribe and publish command messages from/to the message bus.
 
+## EdgeX Service (Internal) Message Bus Requests
+
+Application services (or other EdgeX services in the future) may want to also use message communications to make command requests.  Application services make command requests today via REST.
+
+In order to support this, the following need to be added:
+
+- The command service will also need an internal request topic and internal response topic prefix configuration to allow internal EdgeX services to make command requests (and query requests).
+
+    ``` TOML
+        [MessageQueue]
+            [InternalMessageQueue]
+            Protocol = "redis"
+            Host = "localhost"
+            Port = 6379
+            Type = "redis"
+            RequestTopicPrefix = "edgex/command/request/"   # for publishing requests to the device service; <device-service>/<device-name>/<command-name>/<method> will be added to this publish topic prefix
+            ResponseTopic = "edgex/command/response/#”      # for subscribing to device service responses
+            InternalRequestCommandTopic = "/command/request/#"     # for subscribing to internal command requests
+            InternalResponseCommandTopicPrefix = "/command/response/"    # for publishing responses back to internal service /<device-name>/<command-name>/<method> will be added to this publish topic prefix
+            InternalRequestQueryTopic = "/commandquery/request"
+            InternalResponseQueryTopic = "/commandquery/response"
+            AuthMode = "usernamepassword"                   # required for redis messagebus (secure or insecure).
+            SecretName = "redisdb"
+    ```
+- A new command message client will need to be created to allow internal services (app services in this instance) to conveniently use the message bus communications with core command.
```

Comment:
Need to mention the Clients configuration will also be expanded to include corresponding topics and `UseMessageBus` flag enabling the new messaging based CommandClient to be created. Something like:
```
[Clients]
  [Clients.core-command]
  UseMessageBus = true
  Protocol = "redis"
  Host = "localhost"
  Port = 6379
  CommandRequestTopicPrefix = "/command/request"  /<device-name>/<command-name>/<method> will be added to this publish topic prefix
  CommandResponseTopic = "/command/response/#"
  CommandQueryRequestTopic = "/commandquery/request"
  CommandQueryResponseTopic = "/commandquery/response"
```